File: row-change-event-channel/src/main/scala/nl/haploid/event/channel/repository/RowChangeEvent.scala
package nl.haploid.event.channel.repository

import javax.persistence.{Column, Entity, GeneratedValue, GenerationType, Id, Table}

@Entity
@Table(name = "row_change_events", schema = "projectx")
class RowChangeEvent {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  var id: Long = _

  @Column(name = "table_name")
  var tableName: String = _

  @Column(name = "row_id")
  var rowId: Long = _
}

File: row-change-event/src/main/java/nl/haploid/event/RowChangeEvent.java
package nl.haploid.event;

public class RowChangeEvent {

	private Long id;

	private String tableName;

	private Long rowId;

	public Long getId() {
		return id;
	}

	public Long getRowId() {
		return rowId;
	}

	public String getTableName() {
		return tableName;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public void setRowId(Long rowId) {
		this.rowId = rowId;
	}

	public void setTableName(String tableName) {
		this.tableName = tableName;
	}
}
